### PR TITLE
ft:buyer view order status(add/get/delete)

### PR DESCRIPTION
**Pull request by emilenfc**
# Description

In this feature, buyer can add create order status ofter he create cart, and he can track order status, Admin can change the order status and send update to the buyer profile

# Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have created a loom video for the new feature or enhancement 
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by shyakadavis — APPROVED** [review]
LGTM 🚀

**Review by SergeInGithub — APPROVED** [review]
LGTM

File: src/database/models/Order.model.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { DataTypes } from "sequelize";
+import { sequelize } from ".";
+
+const Order = sequelize.define(
+  "order",
```
Comment (shyakadavis) [inline src/database/models/Order.model.ts:5]:
```suggestion
  "Order",
```

File: src/database/models/Order.model.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      type: DataTypes.DATE,
+      defaultValue: function () {
+        const now = new Date();
+        const deliveryDate = new Date(now.getTime() + 5 * 24 * 60 * 60 * 1000); // add 5 days  of order in milliseconds
+        return deliveryDate;
+      },
+    },
+  },
+  {
+    freezeTableName: false,
```
Comment (shyakadavis) [inline src/database/models/Order.model.ts:40]:
```suggestion
    modelName:'Order',
```

**Review by shyakadavis — DISMISSED** [review]
LGTM ⚖️

**Review by SergeInGithub — DISMISSED** [review]
LGTM

File: src/controllers/Order.controller.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+import {
+  createOrderService,
+  getOrdersByUserService,
+  clearOrdersService,
+  getOrderStatusService,
+  updatedOrderStatusService,
+  getCartItemsService,
+} from "../services";
+
+// To create order
```
Comment (mucyoalexaxel) [inline src/controllers/Order.controller.ts:11]:
```suggestion
```

File: src/controllers/Order.controller.ts [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        status: 500,
+        success: false,
+        message: "unexpected Error while getting order",
+        error: error,
+      });
+    }
+  }
+};
+
+// clear the orders
```
Comment (mucyoalexaxel) [inline src/controllers/Order.controller.ts:70]:
```suggestion
```

File: src/controllers/Order.controller.ts [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+        status: 500,
+        success: false,
+        message: "Error while canceling order",
+        error: error.message,
+      });
+    }
+  }
+};
+
+// GET request to retrieve order status information
```
Comment (mucyoalexaxel) [inline src/controllers/Order.controller.ts:92]:
```suggestion
```

File: src/database/migrations/20230305181417-create-order-tables.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
+/* eslint-disable no-undef */
```
Comment (mucyoalexaxel) [inline src/database/migrations/20230305181417-create-order-tables.js:1]:
```suggestion
```

File: src/database/models/Order.model.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    },
+    status: {
+      type: DataTypes.ENUM("Pending", "Shipped", "Delivered"),
+      defaultValue: "Pending",
+    },
+    expectedDeliveryDate: {
+      type: DataTypes.DATE,
+      defaultValue: function () {
+        const now = new Date();
+        const deliveryDate = new Date(now.getTime() + 5 * 24 * 60 * 60 * 1000); // add 5 days  of order in milliseconds
```
Comment (mucyoalexaxel) [inline src/database/models/Order.model.ts:34]:
```suggestion
        const deliveryDate = new Date(now.getTime() + 5 * 24 * 60 * 60 * 1000);
```

**Review by mucyoalexaxel — CHANGES_REQUESTED** [review]
@emilenfc kindly address the requested changes

**Review by freddykaberuka — APPROVED** [review]

**Pull request merged**